File: reference/SmartPharmacy-Kotlin/app/src/main/java/com/example/ui/screens/ProductDetailsScreen.kt
package com.example.ui.screens

import android.widget.Toast
import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.verticalScroll
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.*
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.platform.LocalContext
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.example.ui.components.ConfirmationDialog
import com.example.ui.components.CustomButton
import com.example.ui.theme.*
import com.example.ui.viewmodel.PharmacyViewModel

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun ProductDetailsScreen(
    viewModel: PharmacyViewModel,
    medicineId: String,
    onNavigateToEdit: (String) -> Unit,
    onNavigateBack: () -> Unit
) {
    val context = LocalContext.current
    val medicines by viewModel.medicines.collectAsState()

    val medicine = medicines.find { it.id == medicineId }
    var showDeleteDialog by remember { mutableStateOf(false) }

    if (medicine == null) {
        Box(modifier = Modifier.fillMaxSize(), contentAlignment = Alignment.Center) {
            Text("Medicine not found")
        }
        return
    }

    val isExpired = viewModel.isExpired(medicine.expiryDate)
    val isExpiringSoon = viewModel.isExpiringIn30Days(medicine.expiryDate)
    val profit = (medicine.sellingPrice - medicine.purchasePrice).coerceAtLeast(0.0)

    Scaffold(
        topBar = {
            TopAppBar(
                title = { Text("Product Details", fontWeight = FontWeight.Bold) },
                navigationIcon = {
                    IconButton(onClick = onNavigateBack) {
                        Icon(imageVector = Icons.Default.ArrowBack, contentDescription = "Back")
                    }
                },
                actions = {
                    IconButton(onClick = { onNavigateToEdit(medicine.id) }) {
                        Icon(imageVector = Icons.Default.Edit, contentDescription = "Edit", tint = PrimaryBlue)
                    }
                    IconButton(onClick = { showDeleteDialog = true }) {
                        Icon(imageVector = Icons.Default.Delete, contentDescription = "Delete", tint = AlertRed)
                    }
                },
                colors = TopAppBarDefaults.topAppBarColors(containerColor = SurfaceWhite)
            )
        },
        containerColor = SurfaceWhite
    ) { innerPadding ->
        Column(
            modifier = Modifier
                .fillMaxSize()
                .padding(innerPadding)
                .padding(16.dp)
                .verticalScroll(rememberScrollState()),
            verticalArrangement = Arrangement.spacedBy(16.dp)
        ) {
            // Header Card
            Card(
                shape = RoundedCornerShape(16.dp),
                colors = CardDefaults.cardColors(containerColor = PrimaryBlueLight),
                modifier = Modifier.fillMaxWidth()
            ) {
                Row(
                    modifier = Modifier.padding(20.dp),
                    verticalAlignment = Alignment.CenterVertically
                ) {
                    Box(
                        modifier = Modifier
                            .size(64.dp)
                            .clip(CircleShape)
                            .background(SurfaceWhite),
                        contentAlignment = Alignment.Center
                    ) {
                        Icon(
                            imageVector = Icons.Default.Medication,
                            contentDescription = null,
                            tint = PrimaryBlue,
                            modifier = Modifier.size(36.dp)
                        )
                    }

                    Spacer(modifier = Modifier.width(16.dp))

                    Column(modifier = Modifier.weight(1f)) {
                        Text(
                            text = medicine.name,
                            fontSize = 18.sp,
                            fontWeight = FontWeight.Bold,
                            color = TextPrimary
                        )
                        Spacer(modifier = Modifier.height(2.dp))
                        Text(
                            text = medicine.company,
                            fontSize = 14.sp,
                            color = TextSecondary
                        )
                        Spacer(modifier = Modifier.height(4.dp))
                        Surface(
                            shape = RoundedCornerShape(12.dp),
                            color = PrimaryBlue
                        ) {
                            Text(
                                text = medicine.category,
                                fontSize = 11.sp,
                                fontWeight = FontWeight.SemiBold,
                                color = SurfaceWhite,
                                modifier = Modifier.padding(horizontal = 8.dp, vertical = 2.dp)
                            )
                        }
                    }
                }
            }

            // Status Badges Row
            Row(
                modifier = Modifier.fillMaxWidth(),
                horizontalArrangement = Arrangement.spacedBy(10.dp)
            ) {
                // Stock Badge
                Card(
                    shape = RoundedCornerShape(14.dp),
                    colors = CardDefaults.cardColors(
                        containerColor = if (medicine.quantity <= medicine.minStockThreshold) AlertRedLight else SuccessGreenLight
                    ),
                    modifier = Modifier.weight(1f)
                ) {
                    Column(modifier = Modifier.padding(14.dp)) {
                        Text("Current Stock", fontSize = 12.sp, color = TextSecondary)
                        Spacer(modifier = Modifier.height(4.dp))
                        Text(
                            text = "${medicine.quantity} units",
                            fontSize = 18.sp,
                            fontWeight = FontWeight.Bold,
                            color = if (medicine.quantity <= medicine.minStockThreshold) AlertRed else SuccessGreen
                        )
                    }
                }

                // Expiry Badge
                Card(
                    shape = RoundedCornerShape(14.dp),
                    colors = CardDefaults.cardColors(
                        containerColor = when {
                            isExpired -> AlertRedLight
                            isExpiringSoon -> WarningOrangeLight
                            else -> PrimaryBlueLight
                        }
                    ),
                    modifier = Modifier.weight(1f)
                ) {
                    Column(modifier = Modifier.padding(14.dp)) {
                        Text("Expiry Status", fontSize = 12.sp, color = TextSecondary)
                        Spacer(modifier = Modifier.height(4.dp))
                        Text(
                            text = when {
                                isExpired -> "EXPIRED"
                                isExpiringSoon -> "Expiring Soon"
                                else -> medicine.expiryDate
                            },
                            fontSize = 16.sp,
                            fontWeight = FontWeight.Bold,
                            color = when {
                                isExpired -> AlertRed
                                isExpiringSoon -> WarningOrange
                                else -> PrimaryBlue
                            }
                        )
                    }
                }
            }

            // Specification Grid
            Card(
                shape = RoundedCornerShape(16.dp),
                colors = CardDefaults.cardColors(containerColor = SurfaceWhite),
                elevation = CardDefaults.cardElevation(defaultElevation = 1.dp),
                modifier = Modifier
                    .fillMaxWidth()
                    .border(1.dp, SurfaceBorder, RoundedCornerShape(16.dp))
            ) {
                Column(
                    modifier = Modifier.padding(18.dp),
                    verticalArrangement = Arrangement.spacedBy(12.dp)
                ) {
                    Text(
                        text = "Pricing & Batch Information",
                        fontSize = 15.sp,
                        fontWeight = FontWeight.Bold,
                        color = TextPrimary
                    )

                    HorizontalDivider(color = SurfaceBorder)

                    DetailRow(label = "Batch Number", value = medicine.batchNumber)
                    DetailRow(label = "Purchase Price", value = "₹${medicine.purchasePrice}")
                    DetailRow(label = "Selling Price (MRP)", value = "₹${medicine.sellingPrice}")
                    DetailRow(
                        label = "Margin / Profit",
                        value = "₹${String.format("%.2f", profit)}",
                        valueColor = SuccessGreen
                    )
                    DetailRow(label = "Expiry Date", value = medicine.expiryDate)
                    DetailRow(label = "Distributor", value = medicine.distributorName)
                }
            }

            Spacer(modifier = Modifier.height(12.dp))

            Row(
                modifier = Modifier.fillMaxWidth(),
                horizontalArrangement = Arrangement.spacedBy(10.dp)
            ) {
                Button(
                    onClick = { onNavigateToEdit(medicine.id) },
                    shape = RoundedCornerShape(14.dp),
                    colors = ButtonDefaults.buttonColors(containerColor = PrimaryBlue),
                    modifier = Modifier
                        .weight(1f)
                        .height(50.dp)
                ) {
                    Icon(imageVector = Icons.Default.Edit, contentDescription = null)
                    Spacer(modifier = Modifier.width(6.dp))
                    Text("Edit Product")
                }

                OutlinedButton(
                    onClick = { showDeleteDialog = true },
                    shape = RoundedCornerShape(14.dp),
                    colors = ButtonDefaults.outlinedButtonColors(contentColor = AlertRed),
                    modifier = Modifier
                        .weight(1f)
                        .height(50.dp)
                ) {
                    Icon(imageVector = Icons.Default.Delete, contentDescription = null)
                    Spacer(modifier = Modifier.width(6.dp))
                    Text("Delete")
                }
            }
        }
    }

    if (showDeleteDialog) {
        ConfirmationDialog(
            title = "Delete Product?",
            message = "Are you sure you want to remove ${medicine.name} from inventory? This action cannot be undone.",
            onConfirm = {
                viewModel.deleteMedicine(medicine.id)
                Toast.makeText(context, "${medicine.name} deleted", Toast.LENGTH_SHORT).show()
                onNavigateBack()
            },
            onDismiss = { showDeleteDialog = false }
        )
    }
}

@Composable
private fun DetailRow(
    label: String,
    value: String,
    valueColor: Color = TextPrimary
) {
    Row(
        modifier = Modifier.fillMaxWidth(),
        horizontalArrangement = Arrangement.SpaceBetween
    ) {
        Text(text = label, fontSize = 13.sp, color = TextSecondary)
        Text(
            text = value,
            fontSize = 14.sp,
            fontWeight = FontWeight.SemiBold,
            color = valueColor
        )
    }
}

File: lib/screens/product_details_screen.dart
import 'package:flutter/material.dart';
import 'package:provider/provider.dart';
import '../core/theme/app_colors.dart';
import '../core/utils/pharmacy_utils.dart';
import '../providers/pharmacy_provider.dart';
import '../routes/app_routes.dart';
import '../widgets/common/confirmation_dialog.dart';
import '../widgets/common/detail_row.dart';

class ProductDetailsScreen extends StatelessWidget {
  final String medicineId;

  const ProductDetailsScreen({
    super.key,
    required this.medicineId,
  });

  @override
  Widget build(BuildContext context) {
    final provider = context.watch<PharmacyProvider>();
    final medicine = provider.medicines.firstWhere(
      (m) => m.id == medicineId,
      orElse: () => throw Exception("Medicine not found"),
    );

    final isExpired = PharmacyUtils.isExpired(medicine.expiryDate);
    final isExpiringSoon = PharmacyUtils.isExpiringIn30Days(medicine.expiryDate);
    final profit = (medicine.sellingPrice - medicine.purchasePrice).clamp(0.0, double.infinity);

    return Scaffold(
      backgroundColor: AppColors.surfaceWhite,
      appBar: AppBar(
        backgroundColor: AppColors.surfaceWhite,
        elevation: 0,
        title: const Text(
          "Product Details",
          style: TextStyle(
            fontWeight: FontWeight.bold,
            color: AppColors.textPrimary,
          ),
        ),
        leading: IconButton(
          icon: const Icon(Icons.arrow_back, color: AppColors.textPrimary),
          onPressed: () => Navigator.pop(context),
        ),
        actions: [
          IconButton(
            icon: const Icon(Icons.edit, color: AppColors.primaryBlue),
            onPressed: () => Navigator.pushNamed(
              context,
              AppRoutes.editProduct,
              arguments: medicine.id,
            ),
          ),
          IconButton(
            icon: const Icon(Icons.delete, color: AppColors.alertRed),
            onPressed: () => _showDeleteDialog(context, provider, medicine.name),
          ),
        ],
      ),
      body: SingleChildScrollView(
        padding: const EdgeInsets.all(16.0),
        child: Column(
          children: [
            // Header Card
            _buildHeaderCard(medicine),
            const SizedBox(height: 16),

            // Status Badges Row
            Row(
              children: [
                Expanded(
                  child: _buildStatusBadge(
                    title: "Current Stock",
                    value: "${medicine.quantity} units",
                    isAlert: medicine.quantity <= medicine.minStockThreshold,
                  ),
                ),
                const SizedBox(width: 10),
                Expanded(
                  child: _buildExpiryBadge(
                    medicine.expiryDate,
                    isExpired,
                    isExpiringSoon,
                  ),
                ),
              ],
            ),
            const SizedBox(height: 16),

            // Specification Grid Card
            _buildSpecificationCard(medicine, profit),
            const SizedBox(height: 24),

            // Action Buttons
            Row(
              children: [
                Expanded(
                  child: ElevatedButton.icon(
                    onPressed: () => Navigator.pushNamed(
                      context,
                      AppRoutes.editProduct,
                      arguments: medicine.id,
                    ),
                    icon: const Icon(Icons.edit),
                    label: const Text("Edit Product"),
                    style: ElevatedButton.styleFrom(
                      backgroundColor: AppColors.primaryBlue,
                      foregroundColor: Colors.white,
                      padding: const EdgeInsets.symmetric(vertical: 14),
                      shape: RoundedRectangleBorder(
                        borderRadius: BorderRadius.circular(14),
                      ),
                    ),
                  ),
                ),
                const SizedBox(width: 10),
                Expanded(
                  child: OutlinedButton.icon(
                    onPressed: () => _showDeleteDialog(context, provider, medicine.name),
                    icon: const Icon(Icons.delete),
                    label: const Text("Delete"),
                    style: OutlinedButton.styleFrom(
                      foregroundColor: AppColors.alertRed,
                      side: const BorderSide(color: AppColors.alertRed),
                      padding: const EdgeInsets.symmetric(vertical: 14),
                      shape: RoundedRectangleBorder(
                        borderRadius: BorderRadius.circular(14),
                      ),
                    ),
                  ),
                ),
              ],
            ),
          ],
        ),
      ),
    );
  }

  Widget _buildHeaderCard(dynamic medicine) {
    return Card(
      elevation: 0,
      margin: EdgeInsets.zero,
      shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(16)),
      color: AppColors.primaryBlueLight,
      child: Padding(
        padding: const EdgeInsets.all(20.0),
        child: Row(
          children: [
            Container(
              width: 64,
              height: 64,
              decoration: const BoxDecoration(
                color: Colors.white,
                shape: BoxShape.circle,
              ),
              child: const Icon(
                Icons.medication,
                color: AppColors.primaryBlue,
                size: 36,
              ),
            ),
            const SizedBox(width: 16),
            Expanded(
              child: Column(
                crossAxisAlignment: CrossAxisAlignment.start,
                children: [
                  Text(
                    medicine.name,
                    style: const TextStyle(
                      fontSize: 18,
                      fontWeight: FontWeight.bold,
                      color: AppColors.textPrimary,
                    ),
                  ),
                  const SizedBox(height: 2),
                  Text(
                    medicine.company,
                    style: const TextStyle(
                      fontSize: 14,
                      color: AppColors.textSecondary,
                    ),
                  ),
                  const SizedBox(height: 8),
                  Container(
                    padding: const EdgeInsets.symmetric(horizontal: 8, vertical: 2),
                    decoration: BoxDecoration(
                      color: AppColors.primaryBlue,
                      borderRadius: BorderRadius.circular(12),
                    ),
                    child: Text(
                      medicine.category,
                      style: const TextStyle(
                        fontSize: 11,
                        fontWeight: FontWeight.w600,
                        color: Colors.white,
                      ),
                    ),
                  ),
                ],
              ),
            ),
          ],
        ),
      ),
    );
  }

  Widget _buildStatusBadge({
    required String title,
    required String value,
    required bool isAlert,
  }) {
    return Card(
      elevation: 0,
      margin: EdgeInsets.zero,
      shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(14)),
      color: isAlert ? AppColors.alertRedLight : AppColors.successGreenLight,
      child: Padding(
        padding: const EdgeInsets.all(14.0),
        child: Column(
          crossAxisAlignment: CrossAxisAlignment.start,
          children: [
            const Text(
              "Current Stock",
              style: TextStyle(fontSize: 12, color: AppColors.textSecondary),
            ),
            const SizedBox(height: 4),
            Text(
              value,
              style: TextStyle(
                fontSize: 18,
                fontWeight: FontWeight.bold,
                color: isAlert ? AppColors.alertRed : AppColors.successGreen,
              ),
            ),
          ],
        ),
      ),
    );
  }

  Widget _buildExpiryBadge(String expiryDate, bool isExpired, bool isExpiringSoon) {
    Color bgColor;
    Color textColor;
    String status;

    if (isExpired) {
      bgColor = AppColors.alertRedLight;
      textColor = AppColors.alertRed;
      status = "EXPIRED";
    } else if (isExpiringSoon) {
      bgColor = AppColors.warningOrangeLight;
      textColor = AppColors.warningOrange;
      status = "Expiring Soon";
    } else {
      bgColor = AppColors.primaryBlueLight;
      textColor = AppColors.primaryBlue;
      status = expiryDate;
    }

    return Card(
      elevation: 0,
      margin: EdgeInsets.zero,
      shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(14)),
      color: bgColor,
      child: Padding(
        padding: const EdgeInsets.all(14.0),
        child: Column(
          crossAxisAlignment: CrossAxisAlignment.start,
          children: [
            const Text(
              "Expiry Status",
              style: TextStyle(fontSize: 12, color: AppColors.textSecondary),
            ),
            const SizedBox(height: 4),
            Text(
              status,
              style: TextStyle(
                fontSize: 16,
                fontWeight: FontWeight.bold,
                color: textColor,
              ),
            ),
          ],
        ),
      ),
    );
  }

  Widget _buildSpecificationCard(dynamic medicine, double profit) {
    return Card(
      elevation: 0,
      margin: EdgeInsets.zero,
      shape: RoundedRectangleBorder(
        borderRadius: BorderRadius.circular(16),
        side: const BorderSide(color: AppColors.surfaceBorder),
      ),
      color: AppColors.surfaceWhite,
      child: Padding(
        padding: const EdgeInsets.all(18.0),
        child: Column(
          crossAxisAlignment: CrossAxisAlignment.start,
          children: [
            const Text(
              "Pricing & Batch Information",
              style: TextStyle(
                fontSize: 15,
                fontWeight: FontWeight.bold,
                color: AppColors.textPrimary,
              ),
            ),
            const Divider(height: 24, color: AppColors.surfaceBorder),
            DetailRow(label: "Batch Number", value: medicine.batchNumber),
            DetailRow(label: "Purchase Price", value: "₹${medicine.purchasePrice}"),
            DetailRow(label: "Selling Price (MRP)", value: "₹${medicine.sellingPrice}"),
            DetailRow(
              label: "Margin / Profit",
              value: "₹${profit.toStringAsFixed(2)}",
              valueColor: AppColors.successGreen,
            ),
            DetailRow(label: "Expiry Date", value: medicine.expiryDate),
            DetailRow(label: "Distributor", value: medicine.distributorName),
          ],
        ),
      ),
    );
  }

  void _showDeleteDialog(
    BuildContext context,
    PharmacyProvider provider,
    String medicineName,
  ) {
    showDialog(
      context: context,
      builder: (ctx) => ConfirmationDialog(
        title: "Delete Product?",
        message:
            "Are you sure you want to remove $medicineName from inventory? This action cannot be undone.",
        onConfirm: () {
          provider.deleteMedicine(medicineId);
          ScaffoldMessenger.of(context).showSnackBar(
            SnackBar(content: Text("$medicineName deleted")),
          );
          Navigator.pop(context);
        },
        onDismiss: () => {},
      ),
    );
  }
}
